core/src/main/scala/kafka/server/ClientRequestQuotaManager.scala

```diff
@@ -18,40 +18,45 @@ package kafka.server
 
 import java.util.concurrent.TimeUnit
 
+import kafka.network.RequestChannel
 import org.apache.kafka.common.MetricName
 import org.apache.kafka.common.metrics._
 import org.apache.kafka.common.utils.Time
 
 
 class ClientRequestQuotaManager(private val config: ClientQuotaManagerConfig,
-                         private val metrics: Metrics,
-                         private val time: Time) extends ClientQuotaManager(config, metrics, QuotaType.Request, time) {
+                                private val metrics: Metrics,
+                                private val time: Time) extends ClientQuotaManager(config, metrics, QuotaType.Request, time) {
   val maxThrottleTimeMs = TimeUnit.SECONDS.toMillis(this.config.quotaWindowSizeSeconds)
   def exemptSensor = getOrCreateSensor(exemptSensorName, exemptMetricName)
 
   def recordExempt(value: Double) {
     exemptSensor.record(value)
   }
 
-  def maybeRecordAndThrottle(sanitizedUser: String, clientId: String, requestThreadTimeNanos: Long,
-      sendResponseCallback: Int => Unit, recordNetworkThreadTimeCallback: (Long => Unit) => Unit): Unit = {
+  def maybeRecordAndThrottle(request: RequestChannel.Request, sendResponseCallback: Int => Unit): Unit = {
```

Comment:
I guessed that this was an attempt to avoid coupling the quota manager with the `Request` object, but it meant a kinda messy signature which was already somewhat tailored to the needs of `Request`. I can revert this change if it seems like the wrong direction.